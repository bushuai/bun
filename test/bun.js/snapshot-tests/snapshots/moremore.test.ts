class Number2 extends Number {
  constructor(value) {
    super(value);
  }
}
class Number3 extends Number2 {
  constructor(value) {
    super(value);
  }
}

class Boolean2 extends Boolean {
  constructor(value) {
    super(value);
  }
}

class Boolean3 extends Boolean2 {
  constructor(value) {
    super(value);
  }

  false = true;

  helloBoolean3() {
    return "true";
  }
}

test("debug", () => {
  expect(1).toMatchSnapshot();
  expect(NaN).toMatchSnapshot();
  expect(Infinity).toMatchSnapshot();
  expect(-Infinity).toMatchSnapshot();
  expect(new Number(1)).toMatchSnapshot();
  expect(new Number2(1)).toMatchSnapshot();
  expect(new Number3(1)).toMatchSnapshot();
  expect(123348923.2341281).toMatchSnapshot();
  expect(false).toMatchSnapshot();
  expect(true).toMatchSnapshot();
  expect(new Boolean(false)).toMatchSnapshot();
  expect(new Boolean(true)).toMatchSnapshot();
  expect(new Boolean2(true)).toMatchSnapshot();
  expect(new Boolean2(false)).toMatchSnapshot();
  expect(new Boolean3(true)).toMatchSnapshot();
  expect(new Boolean3(false)).toMatchSnapshot();
});
