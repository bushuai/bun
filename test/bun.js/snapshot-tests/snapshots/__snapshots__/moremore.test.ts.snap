// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debug 1`] = `1`;

exports[`debug 2`] = `NaN`;

exports[`debug 3`] = `Infinity`;

exports[`debug 4`] = `-Infinity`;

exports[`debug 5`] = `Number {}`;

exports[`debug 6`] = `Number2 {}`;

exports[`debug 7`] = `Number3 {}`;

exports[`debug 8`] = `123348923.2341281`;

exports[`debug 9`] = `false`;

exports[`debug 10`] = `true`;

exports[`debug 11`] = `Boolean {}`;

exports[`debug 12`] = `Boolean {}`;

exports[`debug 13`] = `Boolean2 {}`;

exports[`debug 14`] = `Boolean2 {}`;

exports[`debug 15`] = `
Boolean3 {
  "false": true,
}
`;

exports[`debug 16`] = `
Boolean3 {
  "false": true,
}
`;
